### PR TITLE
Compactor & media recording

### PR DESCRIPTION
**Pull request by lazedo**
added couch compactor. please note that this will be moved into tasks app in the future.
media recording changes to support callflows that already exist with the media url set.
some bug killing and some log messages

### CODE REVIEW

File: applications/crossbar/priv/api/swagger.json [hunk 1]
```diff
@@ -8242,16 +8242,66 @@
...
+                    "default": 60000,
+                    "description": "kazoo_couch sleep between compaction",
+                    "type": "integer"
+                },
+                "sleep_between_poll": {
+                    "default": 3000,
+                    "description": "kazoo_couch sleep between poll",
+                    "type": "integer"
+                },
+                "sleep_between_views": {
```
Comment (fenollp) [inline applications/crossbar/priv/api/swagger.json]: Could the flag mention even the abbreviated time unit? d, s, ms, ...
For Boolean values staring the flag with "should" is also nice

File: core/kazoo_couch/src/kazoo_couch_maintenance.erl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        ,compact_all/0
+        ,compact_node/1
+        ,compact_db/1
+        ,compact_db/2
+
+        ]).
+
+
+start_auto_compaction() ->
```
Comment (fenollp) [inline core/kazoo_couch/src/kazoo_couch_maintenance.erl]: Specs pretty please

File: make/deps.mk [hunk 3]
```diff
@@ -15,7 +15,11 @@ dep_detergent = git https://github.com/pap/detergent e86dfeded3e4f9f3f9278c6a1ae
 dep_jiffy = hex 0.14.7
 dep_jesse = git https://github.com/for-GET/jesse 1.4.0
 dep_nklib = git https://github.com/NetComposer/nklib
-dep_couchbeam = git https://github.com/benoitc/couchbeam 1.3.1
+
+dep_couchbeam = git https://github.com/lazedo/couchbeam ccc80359707ce386fec6ea9a42803369d33f089a
+###dep_couchbeam = git https://github.com/benoitc/couchbeam 1.3.1
+### waiting for pull request
```
Comment (fenollp) [inline make/deps.mk]: Link to or maybe?

File: core/kazoo_attachments/src/kz_att_http.erl [hunk 4]
```diff
@@ -19,14 +19,16 @@
...
 -spec put_attachment(kz_data:connection(), ne_binary(), ne_binary(), ne_binary(), ne_binary(), kz_data:options()) -> any().
 put_attachment(Params, DbName, DocId, AName, Contents, Options) ->
-    #{url := BaseUrl, verb := Verb} = Params,
+    #{url := BaseUrlParam, verb := Verb} = Params,
     {'ok', JObj} = kz_datamgr:open_cache_doc(DbName, DocId, Options),
     Args = [{<<"attachment">>, AName}
            ,{<<"id">>, DocId}
            ],
     Fields = maps:get(field_list, Params, default_format()),
+    FieldSeparator = maps:get(field_separator, Params, <<"/">>),
```
Comment (mark2600) [inline core/kazoo_attachments/src/kz_att_http.erl]: non ticked atoms

File: core/kazoo_couch/src/kz_couch_compactor.erl [hunk 5]
```diff
@@ -0,0 +1,1621 @@
...
+                   {'req_compact_db', ne_binary()} |
+                   {'req_compact_db', ne_binary(), ne_binary()}.
+
+-type not_compacting() :: {'error', 'compactor_down'}.
+-type compactor_heuristic() :: ?HEUR_NONE | ?HEUR_RATIO.
+
+-type node_with_options() :: {ne_binary(), kz_proplist()}.
+-type nodes_with_options() :: [node_with_options()].
+-record(state, {
+          nodes :: ne_binaries() | nodes_with_options()
```
Comment (mark2600) [inline core/kazoo_couch/src/kz_couch_compactor.erl]: indentation is weird here

File: core/kazoo_couch/src/kz_couch_compactor.erl [hunk 6]
```diff
@@ -0,0 +1,1621 @@
...
+        'false' -> {'error', 'compactor_down'}
+    end.
+
+-spec start_auto_compaction() -> {'ok', 'already_started'} |
+                                 {'queued', reference()} |
+                                 not_compacting().
+start_auto_compaction() ->
+    case is_compactor_running() of
+        'true' ->
+            case compact_automatically() of
```
Comment (mark2600) [inline core/kazoo_couch/src/kz_couch_compactor.erl]: maybe try to avoid nested case statements ?

File: make/deps.mk (hunk 3 above)
Comment (lazedo) [inline make/deps.mk]: its there

File: applications/crossbar/priv/api/swagger.json (hunk 1 above)
Comment (lazedo) [inline applications/crossbar/priv/api/swagger.json]: @fenollp this is old code, i do not intend to change this.

File: make/deps.mk (hunk 3 above)
Comment (fenollp) [inline make/deps.mk]: Ah missed that (was on my phone)

File: core/kazoo_attachments/src/kz_att_http.erl (hunk 4 above)
Comment (fenollp) [inline core/kazoo_attachments/src/kz_att_http.erl]: meh

**Pull request merged**